hs_core/views/discovery_view.py

```diff
@@ -35,18 +35,32 @@ def form_valid(self, form):
         return self.render_to_response(context)
 
     def get_context_data(self, **kwargs):
-
         context = super(FacetedSearchMixin, self).get_context_data(**kwargs)
-
-        if self.request.session.get('query_changed', True):
-            context.update({'facets': self.queryset.facet_counts()})
-            self.request.session['facets_items'] = self.queryset.facet_counts()
-        else:
-            if self.request.session.get('facets_items', None):
-                context.update({'facets': self.request.session['facets_items']})
-            else:
+        
+        total_results = SearchQuerySet().all().count()
+        # check whether total number of results is set
+        if self.request.session.get('total_results', None):
```

Comment:
I'm a bit concerned to see that we are stashing values concerned with the context of a single search into the user's session (though I see that this was already being done in this file before this ticket came along).  Can someone explain why we're doing this, instead of using Django's caching interface?